src/test/java/edu/harvard/iq/dataverse/api/auth/BearerTokenAuthMechanismTest.java

```diff
@@ -0,0 +1,191 @@
+package edu.harvard.iq.dataverse.api.auth;
+
+import com.nimbusds.oauth2.sdk.ParseException;
+import com.nimbusds.oauth2.sdk.token.BearerAccessToken;
+import edu.harvard.iq.dataverse.UserServiceBean;
+import edu.harvard.iq.dataverse.api.auth.doubles.BearerTokenKeyContainerRequestTestFake;
+import edu.harvard.iq.dataverse.authorization.AuthenticationServiceBean;
+import edu.harvard.iq.dataverse.authorization.UserRecordIdentifier;
+import edu.harvard.iq.dataverse.authorization.providers.oauth2.OAuth2Exception;
+import edu.harvard.iq.dataverse.authorization.providers.oauth2.oidc.OIDCAuthProvider;
+import edu.harvard.iq.dataverse.authorization.users.AuthenticatedUser;
+import edu.harvard.iq.dataverse.authorization.users.User;
+import edu.harvard.iq.dataverse.settings.JvmSettings;
+import edu.harvard.iq.dataverse.util.testing.JvmSetting;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import javax.ws.rs.container.ContainerRequestContext;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Optional;
+
+import static edu.harvard.iq.dataverse.api.auth.BearerTokenAuthMechanism.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class BearerTokenAuthMechanismTest {
+
+    private static final String TEST_API_KEY = "test-api-key";
+
+    private BearerTokenAuthMechanism sut;
+
+    @BeforeEach
+    public void setUp() {
+        sut = new BearerTokenAuthMechanism();
+    }
+
+    @Test
+    @JvmSetting(key = JvmSettings.FEATURE_FLAG, value = "true", varArgs = "api-bearer-auth")
+    public void testFindUserFromRequest_no_token() throws WrappedAuthErrorResponse {
+        sut.authSvc = Mockito.mock(AuthenticationServiceBean.class);
+        sut.userSvc = Mockito.mock(UserServiceBean.class);
```

Comment:
Please move these `setUp()` to make the tests more readable. (Goes as well for all the other occurences...)